src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -211,6 +211,19 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
             }
         }
 
+        public override BoundNode Visit(BoundNode node)
+        {
+            if (_inExpressionLambda &&
+                !(node is BoundConversion) &&
+                node is BoundExpression expr &&
+                expr.Type is TypeSymbol type &&
+                type.IsByRefLikeType)
```

Comment:
We should probably exclude all "restricted" types, not just ref structs.

Reply:
That's a good point. We have an `IsRestrictedType` helper that should check everything.

Reply:
Sounds good